server/routerlicious/packages/protocol-base/src/quorum.ts

```diff
@@ -229,9 +226,57 @@ export class QuorumProposals extends TypedEventEmitter<IQuorumProposalsEvents> i
             throw new Error("Can't propose in disconnected state");
         }
 
-        const deferred = new Deferred<void>();
-        this.localProposals.set(clientSequenceNumber, deferred);
-        return deferred.promise;
+        return new Promise<void>((resolve, reject) => {
+            // The sequence number that our proposal went pending, which the msn must advance past.
+            // If undefined, then it's not sequenced yet.
+            let targetSequenceNumber: number | undefined;
+
+            // To resolve, a proposal goes through two phases:
+            // 1. Sequencing - waiting for the proposal to be ack'd, to learn the target msn value for approval.
+            // 2. Approval - waiting for the msn to advance past the target value.
+            const localProposalSequencedHandler = (sequencedCSN: number, sequenceNumber: number) => {
+                if (sequencedCSN === clientSequenceNumber) {
+                    targetSequenceNumber = sequenceNumber;
+                    this.stateEvents.off("localProposalSequenced", localProposalSequencedHandler);
+                    this.stateEvents.off("disconnected", disconnectedHandler);
+                    this.stateEvents.on("msnChanged", msnChangedHandler);
+                }
+            };
+            const msnChangedHandler = (msn: number) => {
+                if (targetSequenceNumber !== undefined && msn >= targetSequenceNumber) {
+                    resolve();
+                    removeListeners();
+                }
+            };
+
+            // There are two error flows we consider:  disconnect and disposal.
+            // If we get disconnected before the proposal is sequenced, it has one of two possible futures:
+            // 1. We reconnect and see the proposal was sequenced in the meantime.  The promise resolves.
+            // 2. We reconnect and see the proposal was not sequenced in the meantime.  The promise rejects.
+            const disconnectedHandler = () => {
+                if (targetSequenceNumber === undefined) {
+                    this.stateEvents.once("connected", () => {
+                        reject(new Error("Client disconnected without successfully sending proposal"));
```

Comment:
Could you possibly have `targetSequenceNumber` at this point ? (1) above.

Reply:
Yes, good catch :)  Fixed in latest.